pkg/apis/serving/v1beta1/route_validation.go

```diff
@@ -127,13 +127,33 @@ func (tt *TrafficTarget) Validate(ctx context.Context) *apis.FieldError {
 		errs = errs.Also(apis.ErrMissingOneOf(
 			"revisionName", "configurationName"))
 	}
+	return errs
+}
 
+func (tt *TrafficTarget) validateTrafficPercentage(errs *apis.FieldError) *apis.FieldError {
 	// Check that the traffic Percentage is within bounds.
 	if tt.Percent < 0 || tt.Percent > 100 {
 		errs = errs.Also(apis.ErrOutOfBoundsValue(
 			tt.Percent, 0, 100, "percent"))
 	}
+	return errs
+}
 
+func (tt *TrafficTarget) validateLatestRevision(ctx context.Context) *apis.FieldError {
+	var errs *apis.FieldError
+	if apis.IsInSpec(ctx) && tt.LatestRevision != nil {
+		lr := *tt.LatestRevision
+		pinned := tt.RevisionName != ""
+		if pinned == lr {
+			// The senses for whether to pin to a particular revision or
+			// float forward to the latest revision must match.
+			errs = errs.Also(apis.ErrInvalidValue(lr, "latestRevision"))
```

Comment:
this is the only non null return path, so just
`return apis.ErrInvalidValue(...)`.
Remove the decl above
and `return nil` at the very end.

Reply:
Done